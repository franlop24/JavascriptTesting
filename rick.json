{
    "id": 1,
    "name": "Rich Sanchez",
    "status": "Alive",
    "species": "Human",
    "gender": "Male"
}