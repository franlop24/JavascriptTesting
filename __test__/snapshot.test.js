import { getCharacter } from '../snapshot';
import rick from '../rick.json';
import { expect, test } from '@jest/globals';

describe('Es hora de las instantaneas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    // test('Siempre fallará la instantanea', () => {
    //     const user = {
    //         createdAt: new Date(),
    //         id: Math.floor(Math.random() * 20)
    //     };
    //     expect(user).toMatchSnapshot();
    // });

    test('Tenemos una axception dentro del código', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Francisco Lopez"
        };
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });
});