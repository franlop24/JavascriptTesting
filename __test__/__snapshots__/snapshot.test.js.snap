// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallará la instantanea 1`] = `
Object {
  "createdAt": 2021-07-01T23:22:56.674Z,
  "id": 12,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rich Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una axception dentro del código 1`] = `
Object {
  "id": Any<Number>,
  "name": "Francisco Lopez",
}
`;
